serial_windows.go

```diff
@@ -178,12 +179,22 @@ func setCommState(h syscall.Handle, baud int) error {
 	return nil
 }
 
-func setCommTimeouts(h syscall.Handle) error {
+func setCommTimeouts(h syscall.Handle, readTimeout time.Duration) error {
 	var timeouts structTimeouts
 	const MAXDWORD = 1<<32 - 1
-	timeouts.ReadIntervalTimeout = MAXDWORD
-	timeouts.ReadTotalTimeoutMultiplier = MAXDWORD
-	timeouts.ReadTotalTimeoutConstant = MAXDWORD - 1
+	timeoutMs := uint32(readTimeout.Nanoseconds() / 1e6)
```

Comment:
Same comment about uint32 conversion here.

It would be better to keep it as an int64, then do the conversion to milliseconds, then do range checks, then do the conversion to uint32.